config/routes.rb

```diff
@@ -11,6 +11,7 @@
       resources :users, only: %i[show create update destroy]
       resources :tokens, only: %i[create]
       resources :products, only: %i[index show create update destroy]
+      post 'products/:id/transfer', to: 'products#transfer', as: :transfer
```

Comment:
It is clearer to add this inside the resource "products". You can use member for that